src/NodeParser/StringTemplateLiteralNodeParser.ts

```diff
@@ -18,6 +20,17 @@ export class StringTemplateLiteralNodeParser implements SubNodeParser {
         if (node.kind === ts.SyntaxKind.NoSubstitutionTemplateLiteral) {
             return new LiteralType(node.text);
         }
+        if (
```

Comment:
This looks brittle to me. If `extractLiterals` ever changes, we also have to change code here. 

What if instead you move the logic to the loop below. If something fails in `extractLiterals` (maybe we can make the function return some indicator for that), the loop could break and return `new LiteralType(node.text)`.

Reply:
Agree that this seems a bit brittle.

There doesn't seem to be a `text` property on these nodes, so don't think `node.text` makes much sense? What do you think of the latest commit; try-catch around the `extractLiterals` since it throws an `UnknownTypeError` for non-literal types. And then return a `StringType` when we're unable to find a literal type.